### PR TITLE
SWPWA-1386 - Add new options for klarna payment

### PR DESCRIPTION
**Pull request by AleksandrsKondratjevs**
Hello !
While making this task noticed that klarna payment not working on magento 2.4.0 & swpwa 3.1.7.
So at fist please check this PR : https://github.com/scandipwa/klarna-graphql/pull/2 .

In this task main goal was to add 2 new options for klarna payment (Pay now & Pay over time) from existing project and don't break previous pay method (Pay later). On sandbox it is impossible to check if new pay methods are working correctly, so it may be tested only on real project and then fix if some issues appears.

### CODE REVIEW

File: src/app/component/Klarna/Klarna.component.js [hunk 1]
```diff
@@ -80,14 +80,67 @@ export class Klarna extends PureComponent {
             script.parentNode.removeChild(script);
         }
 
+        const klarnaScript = document.createElement('script');
+        klarnaScript.setAttribute('id', KLARNA_SCRIPT_ID);
+        klarnaScript.setAttribute('src', 'https://x.klarnacdn.net/kp/lib/v1/api.js');
+        klarnaScript.async = true;
+        document.head.appendChild(klarnaScript);
```
Comment (alfredsgenkins) [inline src/app/component/Klarna/Klarna.component.js:87]: Does not this break if going to and from checkout multiple times? The script should be removed from head on un-mount, no?

File: src/app/component/Klarna/Klarna.component.js [hunk 2]
```diff
@@ -80,14 +80,67 @@ export class Klarna extends PureComponent {
...
+    }
+
+    loadPaymentMethod(method) {
+        this.setState({ isLoading: true });
+        localStorage.setItem('kl_pm', method);
+        this.renderScript();
+        this.setState({ canShowPaymentSelector: false });
```
Comment (alfredsgenkins) [inline src/app/component/Klarna/Klarna.component.js:94]: It is not a good practice to do multiple setState's in one function. Move to one setState.

**Review by alfredsgenkins — CHANGES_REQUESTED** [review]

**Comment by alfredsgenkins** [comment]
Also, is this expected, that the composer.json has no Kalrna module in it? If yes, then alright :D

**Comment by AleksandrsKondratjevs** [comment]
> Also, is this expected, that the composer.json has no Kalrna module in it? If yes, then alright :D

File: src/app/component/Klarna/Klarna.component.js (hunk 1 above)
Comment (AleksandrsKondratjevs) [reply src/app/component/Klarna/Klarna.component.js:100]: This assume script removes it self, but the problem is that when it is called, it also triggers another script ( klarna-payments-device-recognition ) that is not removing it self. So Just added logic to remove it if payment was shown. Now if we go back then it also removed.

Comment (AleksandrsKondratjevs) [reply src/app/component/Klarna/Klarna.component.js:100]: > Also, is this expected, that the composer.json has no Kalrna module in it? If yes, then alright :D

I think no, without it klarna won't work :'D. So I added last version into composer

**Review by yeegor — APPROVED** [review]
Seems legit

**Comment by yeegor** [comment]
Please resolve linter issues. Will merge as soon as that is done.

**Comment by riha112** [comment]
It look like you missed to update Klarna.container.js files "pay_later" value to be loaded from storage like it was done in component with "localStorage.getItem('kl_pm');"

**Pull request merged**